eo-maven-plugin/src/main/java/org/eolang/maven/rust_project/Names.java

```diff
@@ -0,0 +1,176 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.rust_project;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.cactoos.scalar.IoChecked;
+import org.cactoos.scalar.Sticky;
+import org.cactoos.scalar.Synced;
+import org.eolang.maven.footprint.FtDefault;
+import org.eolang.maven.util.Home;
+
+/**
+ * The class for storing and assigning names to
+ * rust inserts by its location.
+ * It loads names base from target and needs to
+ * be saved for using in other process.
+ * @since 0.30
+ */
+public final class Names {
+
+    /**
+     * Target directory.
+     * @checkstyle MemberNameCheck (7 lines)
+     */
+    private final Path dest;
+
+    /**
+     * All names.
+     */
+    private final IoChecked<ConcurrentHashMap<String, String>> all;
+
+    /**
+     * Prefix for the name.
+     */
+    private final String prefix;
+
+    /**
+     * Ctor.
+     * @param target Directory where to serialize names.
+     * @throws IOException If any issues with IO.
+     */
+    public Names(final Path target) throws IOException {
+        this.dest = target.resolve("names");
+        this.prefix = target.toString().toLowerCase(Locale.ENGLISH).replaceAll("[^a-z0-9]", "x");
+        this.all = Names.checked(this.dest);
+    }
+
+    /**
+     * Assign the new name to the function.
+     * @param loc Location of the insert.
+     * @return The name.
+     */
+    public String name(final String loc) {
+        final Map<String, String> cached;
+        try {
+            cached = this.all.value();
+        } catch (final IOException exc) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Cannot load names correctly from %s",
+                    this.dest
+                ),
+                exc
+            );
+        }
+        return cached.computeIfAbsent(
+            loc,
+            key -> String.format(
+                "%s%d",
+                this.prefix,
+                cached.size()
+            )
+        );
+    }
+
+    /**
+     * Saves the function to name dispatching table.
+     * @throws IOException If any issues with IO.
+     */
+    public void save() throws IOException {
+        Files.createDirectories(this.dest.getParent());
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final ObjectOutputStream oos = new ObjectOutputStream(baos);
+        oos.writeObject(this.all.value());
+        oos.flush();
+        new Home(this.dest.getParent()).save(
+            new String(Base64.getEncoder().encode(baos.toByteArray())),
```

Comment:
@levBagryansky It's better to specify charset here:
```
 new String(Base64.getEncoder().encode(baos.toByteArray()), StandardCharsets.UTF_8)
```

Reply:
@volodya-lombrozo this method is not overloaded with second argument

Reply:
@levBagryansky It's `String` constructor